pkg/rancher-desktop/main/diagnostics/limaDarwin.ts

```diff
@@ -0,0 +1,57 @@
+import semver from 'semver';
+
+import { DiagnosticsCategory, DiagnosticsChecker } from './types';
+
+import mainEvents from '@pkg/main/mainEvents';
+import { spawnFile } from '@pkg/utils/childProcess';
+import Logging from '@pkg/utils/logging';
+
+const console = Logging.diagnostics;
+
+let kubernetesMemory = Number.POSITIVE_INFINITY;
+
+mainEvents.on('settings-update', (cfg) => {
+  kubernetesMemory = cfg.kubernetes.memoryInGB;
+});
+
+/**
+ * CheckLimaDarwin version checks for compatibility with the host machine.
+ */
+const CheckLimaDarwin: DiagnosticsChecker = {
+  id:       'LIMA_DARWIN_VERSION',
+  category: DiagnosticsCategory.ContainerEngine,
+  applicable() {
+    const isDarwin = process.platform === 'darwin';
+    const isArm = process.arch.startsWith('arm');
```

Comment:
This does not work while our Electron process runs on Rosetta.

Untested (will wait for CI build):

```suggestion
    const isArm = Electron.app.runningUnderARM64Translation || os.arch() === 'arm64';
```